lua/astrocommunity/pack/rust/init.lua

```diff
@@ -1,21 +1,63 @@
-local utils = require "astronvim.utils"
 return {
   { import = "astrocommunity.pack.toml" },
   {
     "nvim-treesitter/nvim-treesitter",
+    optional = true,
     opts = function(_, opts)
       if opts.ensure_installed ~= "all" then
-        opts.ensure_installed = utils.list_insert_unique(opts.ensure_installed, "rust")
+        opts.ensure_installed = require("astrocore").list_insert_unique(opts.ensure_installed, "rust")
       end
     end,
   },
   {
-    "simrat39/rust-tools.nvim",
-    ft = { "rust" },
-    init = function() astronvim.lsp.skip_setup = utils.list_insert_unique(astronvim.lsp.skip_setup, "rust_analyzer") end,
+    "AstroNvim/astrolsp",
+    ---@param opts AstroLSPOpts
+    opts = {
+      handlers = { rust_analyzer = false }, -- disable setup of `rust_analyzer`
+      ---@diagnostic disable: missing-fields
+      config = {
+        rust_analyzer = {
+          settings = {
+            ["rust-analyzer"] = {
+              checkOnSave = {
+                command = "clippy",
+              },
+              assist = {
+                importEnforceGranularity = true,
+                importPrefix = "crate",
+              },
+              completion = {
+                postfix = {
+                  enable = false,
+                },
+              },
+              inlayHints = {
+                lifetimeElisionHints = {
+                  enable = true,
+                  useParameterNames = true,
+                },
+              },
+            },
+          },
+        },
+      },
+    },
+  },
+  {
+    "jay-babu/mason-nvim-dap.nvim",
+    optional = true,
+    opts = function(_, opts)
+      opts.ensure_installed = require("astrocore").list_insert_unique(opts.ensure_installed, "codelldb")
+    end,
+  },
+  {
+    "mrcjkb/rustaceanvim",
+    version = "^3",
+    ft = "rust",
     opts = function()
       local adapter
       local success, package = pcall(function() return require("mason-registry").get_package "codelldb" end)
+      local cfg = require "rustaceanvim.config"
```

Comment:
rustaceanvim auto-detects `codelldb` from the path and from the mason-registry (if installed).
So this should now be redundant :smile:

Reply:
Interesting...I'll remove it! :D

Reply:
Doesn't this mean that we can remove the whole opts function? @mrcjkb

Reply:
I'm not sure what this part does:

```lua
return { server = require("astrolsp").lsp_opts "rust_analyzer", dap = { adapter = adapter } }
```

I suppose ` require("astrolsp").lsp_opts` includes an `on_attach` function that sets keymaps, etc.?